lib/util.ts

```diff
@@ -33,21 +33,26 @@ export function roundTime(chain: ChainInfo, round: number) {
 
 export type HttpOptions = {
     userAgent?: string
+    headers?: Record<string, string>
 }
 
+// taking a separate `userAgent` param for backwards compatibility
 export const defaultHttpOptions: HttpOptions = {
-    userAgent: `drand-client-${LIB_VERSION}`
+    userAgent: `drand-client-${LIB_VERSION}`,
+    headers: {
+        'Access-Control-Allow-Origin': '*'
+    }
 }
 
 // eslint-disable-next-line  @typescript-eslint/no-explicit-any
 export async function jsonOrError(url: string, options: HttpOptions = defaultHttpOptions): Promise<any> {
-    let requestOptions = {}
+    const headers = {...options.headers}
 
     if (options.userAgent) {
-        requestOptions = {...requestOptions, headers: {['User-Agent']: options.userAgent}}
+        headers['User-Agent'] = options.userAgent
     }
 
-    const response = await fetch(url, requestOptions)
+    const response = await fetch(url, {headers})
```

Comment:
What if someone wanted to pass other options than headers?
WDYT @webmaster128?

Reply:
happy to add support for this later - optionally we could provide a way for users to pass their own client (would likely be a breaking change though)

Reply:
Looking at non-headers settings for this field, I don't see any need to change them at the moment (https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options)